spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/DiscoveryClient.java

```diff
@@ -53,6 +54,19 @@
 	 */
 	List<String> getServices();
 
+	/**
+	 * Can be used to verify the client is valid and able to make calls.
+	 * <p>
+	 * A successful invocation with no exception thrown implies the client is able to make
+	 * calls.
+	 * <p>
+	 * The default implementation simply calls {@link #getServices()} - client
+	 * implementations can override with a lighter weight operation if they choose to.
+	 */
+	default void probe() {
```

Comment:
Let's remove this part and simply have `getServices()` as optional based on the propery

Reply:
If we do that then the `DiscoveryClientHealthIndicator ` will not really be verifying the connectivity of the client(s) as it will not be invoking the client.